mathics/builtin/forms/output.py

```diff
@@ -1124,3 +1129,23 @@ def eval_makeboxes_matrix(self, table, f, evaluation, options):
             return RowBox(String("("), result, String(")"))
 
         return result
+
+
+# Custom private symbol that helps DisplayForm to control if
+# MakeBoxes must process BoxElements as expressions, or keep as they are.
+class PrivateSymbolThatControlsTheStateOfDisplayForm(Builtin):
+    """
+    <dl>
+    <dt>'PrivateSymbolThatControlsTheStateOfDisplayForm'
```

Comment:
Small thing - please indent nested `<dt>`.

Reply:
@rocky, the question is, do you think that this is a better approach?

Reply:
@mmatera good question. I have to think about it. What I'd be looking for is clues in the WMA code. 

Think of this like the way a physicist would. We have two hypotheses for how to code. Is there a test that distinguishes these? Suppose we hypothesize that that WMA follows the WMA way to do this, what test could we come up with that would distinguish the implementations?

Reply:
Ja, it is a curious request! Because it is exactly as this physicist is trying to distinguish what is the implementation in WMA :) 

Look at this, for example,

![imagen](https://user-images.githubusercontent.com/2862288/201536208-4a93d3d9-de2f-46ae-94aa-a5e5f25ba10e.png)

If I believe in this output, I have to assume that there is not a private hidden symbol but an internal variable inside the kernel.

Reply:
Ok - I trust your conclusion, but I don't follow it. The output above doesn't indicate anything to me one way or another. Please break it down for a novice. 

If it were implemented as a private hidden symbol what then would you expect to see?

Reply:
We should probably implement TracePrint so that we can check correspondence easier.

Reply:
I would expect an evaluation for the corresponding hidden symbol. This is an example in which internal variables are used:

![imagen](https://user-images.githubusercontent.com/2862288/201537236-94283025-9db2-4811-8a75-00d5fde18349.png)

(I just cut the first lines)

Notice the use of symbols in hidden contexts (BoxForm`ls, IntegrateDump`TestLimits, BoxForm`opt). Tracing `MakeBoxes[DisplayForm[...]]`  I do not see such kind of symbols in hidden contexts.